nvi-evaluator/src/main/java/no/sikt/nva/nvi/evaluator/calculator/NviCalculator.java

```diff
@@ -137,18 +139,39 @@ private static URI createUri(String affiliation) {
                    .getUri();
     }
 
+    private static boolean isHttpOk(HttpResponse<String> response) {
+        return response.statusCode() == HttpURLConnection.HTTP_OK;
+    }
+
+    private static boolean isNotFound(HttpResponse<String> response) {
+        return response.statusCode() == HttpURLConnection.HTTP_NOT_FOUND;
+    }
+
+    private static boolean isSuccessOrNotFound(HttpResponse<String> response) {
+        return isHttpOk(response) || isNotFound(response);
+    }
+
+    private static Boolean getNviValue(HttpResponse<String> response) {
+        return Optional.ofNullable(response.body())
+                   .map(NviCalculator::toCustomer)
+                   .map(CustomerResponse::nviInstitution)
+                   .orElse(false);
+    }
+
     private List<String> fetchNviInstitutions(List<String> affiliationUris) {
         return affiliationUris.stream()
                    .filter(this::isNviInstitution)
                    .collect(Collectors.toList());
     }
 
-    private boolean isNviInstitution(String affiliation) {
-        return attempt(() -> uriRetriever.getRawContent(createUri(affiliation), CONTENT_TYPE))
-                   .map(Optional::get)
-                   .map(NviCalculator::toCustomer)
-                   .map(CustomerResponse::nviInstitution)
-                   .orElseThrow(new RuntimeException(COULD_NOT_FETCH_AFFILIATION_MESSAGE + affiliation));
+    private Boolean isNviInstitution(String affiliation) {
+        var response = uriRetriever.fetchResponse(createUri(affiliation), CONTENT_TYPE);
+        if (response.isPresent() && isSuccessOrNotFound(response.get())) {
+            LOGGER.info(AFFILIATION_FETCHED_SUCCESSFULLY_MESSAGE, response.get().statusCode());
+            return getNviValue(response.get());
```

Comment:
You can write this with uriRetriever.fetchResponse(....).filter(this::isSuccessOrNotFound).map(this::getNviValue).orElseThrow(...)